### DISCUSSION TITLE
【新提案】逆向迭代Map、Set等目前仅供迭代访问的原生数据结构

### DISCUSSION THREAD

**Discussion by LongTengDao** [JSCIG/es-discuss#48] [Ideas]
时不时会遇到需要逆向消费Map的场景，整个解构为数组再逆向，时空开销太大。

当然了，我不太了解底层实现，这个提案是建立在这并不会导致严重的底层实现的性能浪费的前提下的。
否则这毕竟是相对小概率的场景，实际需要时自行解构为数组，当然好过一律支付为了兼容新功能而浪费的底层索引开销（比如把单向链表变成双向链表，多存储了一倍的索引）。

**Comment by hax** [comment]
如果你是说 `new Set([1,2,3])` 按照 3, 2, 1 来迭代，那么双端迭代或 reverseIterator 提案能满足。不过双端迭代提案在上一次陈述的时候遇到比较大的阻力，我暂时也不知道怎么推进。

**Reply by LongTengDao** [reply]
啊，之前不知道有 reverseIterator 提案。这个感觉应该会比较容易推进吧？
